src/service/app.py

```diff
@@ -60,6 +64,37 @@ async def build_app_wrapper():
 
 async def build_app(app, cfg):
     app.state.kb_auth = await kb_auth.create_auth_client(cfg.auth_url, cfg.auth_full_admin_roles)
+    cli = aioarango.ArangoClient(hosts=cfg.arango_url)
+    if cfg.create_db_on_startup:
+        sysdb = await _get_arango_db(cli, "_system", cfg)
+        try:
+            await sysdb.create_database(cfg.arango_db)
+        except aioarango.exceptions.DatabaseCreateError as e:
+            if e.error_code != 1207:  # duplicate name error, ignore, db exists
+                raise
+    db = await _get_arango_db(cli, cfg.arango_db, cfg)
+    app.state.storage = await create_storage(
+        db, create_collections_on_startup=cfg.create_db_on_startup)
+
+
+_BACKOFF = [0, 1, 2, 5, 10, 30]
+
+
+async def _get_arango_db(cli: aioarango.ArangoClient, db: str, cfg: CollectionsServiceConfig):
+    for t in _BACKOFF:
+        if t > 0:
+            print(f"Waiting for {t}s and retrying db connection")
+            time.sleep(t)
+        try:
+            if cfg.arango_user:
+                rdb = await cli.db(
+                    db, verify=True, username=cfg.arango_user, password=cfg.arango_pwd)
+            else:
+                rdb = await cli.db(db, verify=True)
```

Comment:
that's for some local testing scenarios? Should we raise if user or pwd is missing?

Reply:
i suppose docker-compose is local testing...

Reply:
okay. cool 👍